src/SFML/Audio/SoundStream.cpp

```diff
@@ -82,7 +82,11 @@ void SoundStream::initialize(unsigned int channelCount, unsigned int sampleRate)
     m_channelCount = channelCount;
     m_sampleRate = sampleRate;
     m_samplesProcessed = 0;
-    m_isStreaming = false;
+
+    {
+        std::scoped_lock lock(m_threadMutex);
+        m_isStreaming = false;
+    }
```

Comment:
Why do we need a lock here? What are we synchronizing against?

Reply:
The write to `m_isStreaming` is not guaranteed to be visible to reads from another thread started after this call returns.